doc/specs/#2046 - Unified keybindings and commands, and synthesized action names.md

```diff
@@ -0,0 +1,589 @@
+  attempt to use `GenerateName` to create the unique string for the action
+  associated with this command. If the command does have a `name` set, we'll use
+  that string instead, allowing the user to override the default name.
+  - If a command has it's name set to `null`, then we'll ignore the command
+    entirely, not just use the generated name.
+
+[**Appendix 1**](#appendix-1-name-generation-samples-for-ShortcutActions) below
+shows a complete sample of the strings that will be generated for each of the existing
+`ShortcutActions`, and many of the actions that have been proposed, but not yet
+implemented.
+
+These strings should be human-friendly versions of the actions and their
+associated args. For some of these actions, with very few arguments, the strings
+can be relatively simple. Take for example, `CopyText`:
+
+JSON | Generated String
+-- | --
+`{ "action":"copyText" }` | "Copy text"
+`{ "action":"copyText", "singleLine": true }` | "Copy text as a single line"
+`{ "action":"copyText", "singleLine": true, "copyFormatting": true }` | "Copy text without formatting"
+`{ "action":"copyText", "singleLine": true, "copyFormatting": true }` | "Copy text as a single line without formatting"
+
+CopyText is a bit of a simplistic case however, with very few args or
+permutations of argument values. For things like `newTab`, `splitPane`, where
+there are many possible arguments and values, it will be acceptable to simply
+append `", property:value"` strings to the generated names for each of the set
+values.
+
+For example:
+
+JSON | Generated String
+-- | --
+`{ "action":"newTab", "profile": "Hello" }` | "Open a new tab, profile:Hello"
+`{ "action":"newTab", "profile": "Hello", "directory":"C:\\", "commandline": "wsl.exe", title": "Foo" }` | "Open a new tab, profile:Hello, directory:C:\\, commandline:wsl.exe, title:Foo"
+
+
+This is being chosen in favor of something that might be more human-friendly,
+like "Open a new tab with profile {profile name} in {directory} with
+{commandline} and a title of {title}". This string would be much harder to
+synthesize, especially considering localization concerns.
+
+#### Remove the resource key notation
+
+Since we'll be using localized names for each of the actions in `GenerateName`,
+we no longer _need_ to provide the `{ "name":{ "key": "SomeResourceKey" } }`
+syntax introduced in the original spec. This functionality was used to allow us
+to define localizable names for the default commands.
+
+However, I think we should keep this functionality, to allow us additional
+flexibility when defining default commands.
+
+### Complete Defaults
+
+Considering both of the above proposals, the default keybindings and commands
+will be defined as follows:
+
+* The current default keybindings will be untouched. These actions will
+  automatically be added to the Command Palette, using their names generated
+  from `GenerateName`.
+  - **TODO: FOR DISCUSSION**: Should we manually set the names for the default

```

Comment:
As mentioned above, `index` should be resolved to be more descriptive for what it points to. This will be particularly important for https://github.com/microsoft/terminal/pull/5888